controller/controller.go

```diff
@@ -167,23 +166,11 @@ func (c *Controller) DeleteFunc(obj interface{}) {
 
 	resourceSet, err := c.resourceSet(obj)
 	if IsNoResourceSet(err) {
-		// In case the resource router is not able to find any resource set to
-		// handle the reconciled runtime object, we stop here. Note that we just
-		// remove the finalizer regardless because at this point there will never be
-		// a chance to remove it otherwhise because nobody wanted to handle this
-		// runtime object anyway.
-
-		c.logger.Log("level", "debug", "message", "removing finalizer from runtime object")
-
-		err = c.removeFinalizer(obj)
-		if err != nil {
-			c.logger.Log("level", "error", "message", "stop reconciliation due to error", "stack", fmt.Sprintf("%#v", err))
-			return
-		}
-
-		c.logger.Log("level", "debug", "message", "removed finalizer from runtime object")
-
```

Comment:
This feels bad to me. Like leaving the doors open to have orphaned resources.

Reply:
So don't we have an deadlock now? We should also have tests for this BTW. I think this change will break the operators and we do not have automated tests to verify this. I remember we added this handling for a reason.

Reply:
Ack. This is for deadlock prevention. I will restore it. I want to have logs for now.

Reply:
Reverted.